runtime/jvmti/jvmtiCapability.c

```diff
@@ -378,6 +378,12 @@ jvmtiAddCapabilities(jvmtiEnv* env,
 			 * Set it to 512KB which is default sampling interval as per JEP 331 specification.
 			 */
 			vm->memoryManagerFunctions->j9gc_set_allocation_sampling_interval(currentThread, 512 * 1024);
+			
+			if (J9_ARE_ANY_BITS_SET(jvmtiData->flags, J9JVMTI_FLAG_SAMPLED_OBJECT_ALLOC_ENABLED)) {
```

Comment:
This check should occur before the call to `j9gc_set_allocation_sampling_interval`.  An agent that can't get the capability shouldn't be able to modify the sampling interval.

Reply:
Agreed, the second agent might fail to call `jvmtiAddCapabilities()` and exit gracefully, but the first agent along with its sampling interval and JVM itself should continue without impact.
Fixed the order, please review.